Backend/server.js

```diff
@@ -58,6 +58,40 @@ app.get('/getCourses', async (req, res) => {
   }
 });
 
+async function cashSubmissions(userId, courses) {
+  var gainz = 0;
+
+  courses.map((course) => {
+    course[3].map((submission) => {
+      var multiplier = 1;
+      //multiplier *= WEIGHT_LOGIC
+      if (submission[7] && submission[4]) { //score
+        multiplier *= (10*submission[7] / 8 / submission[4]);
+      }
+      if (submission[3] && submission[6]) { //due date
+        var due = new Date(submission[3]);
+        due = due.getTime();
+        var sub = new Date(submission[6]);
+        sub = sub.getTime();
+        var unlock;
+        if (submission[2]) { //unlock date given
+          unlock = new Date(submission[2]);
+        } else { // assume it unlocks 4 weeks before it's due
+          unlock = due - 4 * 604800000;
+        }
+        const frac = (due - sub) / (due - unlock);
+        multiplier *= Math.min((Math.pow(frac, 1/3) + .5), 1.5);
```

Comment:
`frac` being negative (late submissions) will cause this to become `NaN` since `Math.pow` can't handle this. `Math.cbrt` won't have an error, but do we really want to reward late submissions?

Reply:
Cube root should account for lateness - I'll use it and bound it so users can't lose gems at least